src/main/java/net/snowflake/client/jdbc/SFAsyncResultSet.java

```diff
@@ -181,7 +167,7 @@ private void getRealResults() throws SQLException {
    */
   @Override
   public boolean next() throws SQLException {
-    getRealResults();
```

Comment:
Discovered that metadata is needed to call findColumns() for get functions using String columnNames, such as getDate("COLA"). This shouldn't affect performance because next() and getMetaData() did almost the same thing anyway by calling getRealResults().